modules/account-lib/src/coin/dot/transactionBuilderFactory.ts

```diff
@@ -1,42 +1,228 @@
+   *
+   * @see https://wiki.polkadot.network/docs/build-transaction-construction
+   */
+  metadataRpc(metadataRpc: string): this {
+    this._metadataRpc = metadataRpc;
+    return this;
+  }
+
+  /**
+   *
+   * The specVersion for transaction format.
+   *
+   * @param {number} specVersion
+   * @returns {TransactionBuilder} This transaction builder.
+   *
+   * @see https://wiki.polkadot.network/docs/build-transaction-construction
+   */
+  specVersion(specVersion: number): this {
+    this._specVersion = specVersion;
+    return this;
+  }
+
+  /**
+   *
+   * The chainName for transaction format.
+   *
+   * @param {number} chainName
+   * @returns {TransactionBuilder} This transaction builder.
+   *
+   * @see https://wiki.polkadot.network/docs/build-transaction-construction
+   */
+  chainName(chainName: string): this {
+    this._chainName = chainName;
+    return this;
   }
 
-  /** @inheritdoc */
-  from(raw: Uint8Array | string): TransactionBuilder {
-    throw new NotImplementedError('from not implemented');
+  /**
+   * Sets the
+   * specName,
+   * genesisHash - genesisHash of the chain,
+   * metadataRpc - The SCALE-encoded metadata for the runtime when submitted,
+   * specVersion - The current spec version for the runtime,
+   * chainName - chainName,
+   * registry,
+   * of the testnet
+   *
+   * @returns {TransactionBuilder} This transaction builder.
+   *
+   * @see https://wiki.polkadot.network/docs/build-transaction-construction
+   */
+  testnet(): this {
+    this.specName('polkadot');
+    this.metadataRpc(testnetMetadataRpc);
+    this.specVersion(9100);
+    this.chainName('Polkadot');
+    this.genesisHash('0x2b8d4fdbb41f4bc15b8a7ec8ed0687f2a1ae11e0fc2dc6604fa962a9421ae349');
+    this.buildRegistry();
+    return this;
   }
 
-  /** @inheritdoc */
-  getWalletInitializationBuilder(tx?: Transaction): WalletInitializationBuilder {
-    return TransactionBuilderFactory.initializeBuilder(tx, new WalletInitializationBuilder(this._coinConfig));
+  private staticsConfig(): void {
+    const networkConfig = this._coinConfig.network as DotNetwork;
+    this.specName(networkConfig.specName);
+    this.genesisHash(networkConfig.genesisHash);
+    this.specVersion(networkConfig.specVersion);
+    this.chainName(networkConfig.chainName);
   }
 
-  /** @inheritdoc */
-  getTransferBuilder(tx?: Transaction): TransferBuilder {
-    return TransactionBuilderFactory.initializeBuilder(tx, new TransferBuilder(this._coinConfig));
+  /**
+   * Sets the
+   * specName,
+   * genesisHash - genesisHash of the chain,
+   * metadataRpc - The SCALE-encoded metadata for the runtime when submitted,
+   * specVersion - The current spec version for the runtime,
+   * chainName - chainName,
+   * registry,
+   * of the westened
+   *
+   * @returns {TransactionBuilder} This transaction builder.
+   *
+   * @see https://wiki.polkadot.network/docs/build-transaction-construction
+   */
+  westend(): this {
```

Comment:
Where is the main difference here between testnet and westend @gibrangul95 . I'm confused looking this on the transactionBuilder also comparing this with the statics changes. That only mention mainnet and testnet.
Example: On the network.ts file on statics I can see this for testnet:
class PolkadotTestnet extends Testnet implements DotNetwork {
  name = 'Westend';
  family = CoinFamily.DOT;
  explorerUrl = 'https://westend.subscan.io/extrinsic/';
  specName = 'westend' as PolkadotSpecNameType;
  genesisHash = '0xe143f23803ac50e8f6f8e62695d1ce9e4e1d68aa36c1cd2cfd15340213f3423e';
  specVersion = 9122;
  chainName = 'Westend';
}

Reply:
testnet() is specifically for a local node, im thinking maybe we can move this to stub builder in the tests. Has nothing to do with westend.

Reply:
@rliberal-simtlix moved to test cases

Reply:
This should be inferred if the used instantiates this builder with westend (PolkadotTestnet) configuration